core/src/main/java/io/grpc/transport/ServerTransportListener.java

```diff
@@ -48,4 +49,9 @@
    */
   ServerStreamListener streamCreated(ServerStream stream, String method,
       Metadata.Headers headers);
+
+  /**
+   * The transport completed shutting down. All resources have been released.
+   */
+  void transportTerminated();
```

Comment:
Actually, I'm not sure how `onClosed()/closed()` has any less ambiguity than `shutdown()` or `terminated()`. I assume you're intending `shutdown` to indicate that you're beginning shutdown and `terminated indicating shutdown has completed?  If that's the case than `stopping` and `stopped` would be more clear. It would seem that `closing` and `closed` would fit just as well.